sklearn/gaussian_process/gpr.py

```diff
@@ -209,12 +222,17 @@ def obj_func(theta, eval_gradient=True):
                         "Multiple optimizer restarts (n_restarts_optimizer>0) "
                         "requires that all bounds are finite.")
                 bounds = self.kernel_.bounds
-                for iteration in range(self.n_restarts_optimizer):
+                theta_initials = []
+                for i in range(self.n_restarts_optimizer):
                     theta_initial = \
                         self.rng.uniform(bounds[:, 0], bounds[:, 1])
-                    optima.append(
-                        self._constrained_optimization(obj_func, theta_initial,
-                                                       bounds))
+                    theta_initials.append(theta_initial)
+                Parallel(n_jobs=self.n_jobs)(delayed(self._optima_iterations,
+                                                     check_pickle=False)
```

Comment:
It gives an error "TypeError: can't pickle instancemethod objects" when check_pickle=True.
Hence I did otherwise. Please let me know if it is wrong.